congress/votes.py

```diff
@@ -0,0 +1,288 @@
+from .base import Client, SingleListParser, BaseValidator, BaseParser
+
+
+class Votes:
+    """Class for interacting with `Votes` endpoint of the Propublica Congress API
+
+    Parameters
+    ----------
+    api_key : str
+        The Propublica API key to use for authentication
+    """
+
+    def __init__(self, api_key):
+        self._api_key = api_key
+
+    def get_recent_votes(self, chamber, offset=0):
+        """Return the 20 most recent results, sorted by date and roll call number
+
+        Parameters
+        ----------
+        chamber : str
+            `house`, `senate`, `both`
+        offset : int
+            This value offsets the returns of recent results
+
+        Returns
+        -------
+        dict
+            A dictionary containing the json response
+        """
+
+        client = Client(self._api_key,
+                        BaseParser(),
+                        BaseValidator())
+        return client.get(chamber, "votes/recent.json", offset=offset)
+
+    def get_rollcall_votes(self, congress, chamber, ses_no, rcall_no):
+        """
+        Return roll call votes
+
+        Parameters
+        ----------
+        congress : str
+            From 102 to 115 for House, and from 80 to 115 for Senate
+        chamber : str
+            `house`, `senate`
+        ses_no : str
+            Session number: `1` or `2`, depending on year
+            `1` is odd-numbered years, `2` is even-numbered years
+        rcall_no : str
+            Roll call number
+
+        Returns
+        -------
+        dict
+            A dictionary containing the json response
+
+        """
+
+        client = Client(self._api_key,
+                        BaseParser(),
+                        BaseValidator())
+        return client.get(congress, chamber,
+                          "sessions", ses_no, "votes",
+                          "{}.json".format(rcall_no))
+
+    def get_votesbytype(self, congress, chamber, vote_type):
+        """
+        Return votes by type
+
+        Parameters
+        ----------
+        congress : str
+            From 102 to 115 for House, and from 80 to 115 for Senate
+        chamber : str
+            `house`, `senate`
+        vote_type : str
+            `missed`, `party`, `loneno` or `perfect`
+
+        Returns
+        -------
+        pandas.DataFrame
+        """
+        if vote_type not in ['missed', 'party', 'loneno', 'perfect']:
```

Comment:
we should be doing this in the validator, but correct idea.  I recommend writing a validator which checks membership in a set which can probably be used for all of these sorts of errors.